sdk/src/main/scala/com/akkaserverless/javasdk/impl/valueentity/ValueEntitiesImpl.scala

```diff
@@ -164,11 +165,23 @@ final class ValueEntitiesImpl(_system: ActorSystem,
             context.deactivate() // Very important!
           }
 
+          val serializedPrimaryEffect = effect.primaryEffect match {
+            case UpdateState(state) => UpdateState(service.anySupport.encodeJava(state))
+            case DeleteState => DeleteState
+            case NoPrimaryEffect => NoPrimaryEffect
+          }
+          // FIXME something more that needs to be serialized?
+          val serializedSecondaryEffect = effect.secondaryEffect match {
+            case MessageReplyImpl(message, metadata, sideEffects) =>
+              MessageReplyImpl(service.anySupport.encodeJava(message), metadata, sideEffects)
+            case other => other
+          }
```

Comment:
This is something that was causing the integration tests to fail. 

We must convert whatever type we get to PbAny and that was not always the case.

Reply:
Right, so here we choose to do the serialization in the `Impl` rather than in the `Handler`. Let's see how long we can get away with that :)